From ccea985840cf5b05b0359067854abc3da080e737 Mon Sep 17 00:00:00 2001
From: Jan Grulich <jgrulich@redhat.com>
Date: Thu, 13 Dec 2018 07:47:19 +0100
Subject: FileChooser: make use of current_name property in Save dialog

It's not used in Qt portal implementation, but gtk has API for that and sets it in Save dialog
and we should try to call selectFile() with it to get pre-selected file for the dialog

BUG: 402077
---
 src/filechooser.cpp | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/filechooser.cpp b/src/filechooser.cpp
index 37e7744..19f4d42 100644
--- a/src/filechooser.cpp
+++ b/src/filechooser.cpp
@@ -249,11 +249,6 @@ uint FileChooserPortal::SaveFile(const QDBusObjectPath &handle,
     fileDialog->setModal(modalDialog);
     fileDialog->setAcceptMode(QFileDialog::AcceptSave);
 
-    // TODO: Looks Qt doesn't have API for this
-    // if (!currentName.isEmpty()) {
-    //    fileDialog->selectFile(currentName);
-    // }
-
     if (!currentFolder.isEmpty()) {
         fileDialog->setDirectoryUrl(QUrl(currentFolder));
     }
@@ -262,6 +257,10 @@ uint FileChooserPortal::SaveFile(const QDBusObjectPath &handle,
         fileDialog->selectFile(currentFile);
     }
 
+    if (!currentName.isEmpty()) {
+        fileDialog->selectFile(currentName);
+    }
+
     if (!acceptLabel.isEmpty()) {
         fileDialog->setLabelText(QFileDialog::Accept, acceptLabel);
     }
-- 
cgit v1.1
